### PR TITLE
Feature/5 one time sync

### PR DESCRIPTION
**Pull request by ovelartelius**

### CODE REVIEW

**Review by LinusEkstrom — CHANGES_REQUESTED** [review]

File: Addon.Episerver.EnvironmentSynchronizer/InitializationModule/SynchronizationInitializationModule.cs [hunk 1]
```diff
@@ -12,18 +12,34 @@ namespace Addon.Episerver.EnvironmentSynchronizer.InitializationModule
...
+					var store = ServiceLocator.Current.GetInstance<EnvironmentSynchronizationStore>();
+					var flag = store.GetFlag();
+					if (flag != null && flag.Environment == environmentSynchronizationManager.GetEnvironmentName())
+					{
+						runInitialization = false;
+						Logger.Information($"Environment Synchronizer will not run. Flag match the current environment {flag.Environment}");
+					}
+				}
+
+				if (!runInitialization) return;
```
[the remark this answers is not in the document]
Comment (ovelartelius) [reply Addon.Episerver.EnvironmentSynchronizer/InitializationModule/SynchronizationInitializationModule.cs:41]: Agree. I blame ReSharper :)

File: Addon.Episerver.EnvironmentSynchronizer/Properties/AssemblyInfo.cs [hunk 2]
```diff
@@ -32,4 +31,4 @@
 // You can specify all the values or you can default the Build and Revision Numbers
 // by using the '*' as shown below:
 // [assembly: AssemblyVersion("1.0.*")]
-[assembly: AssemblyVersion("0.1.0.*")]
+[assembly: AssemblyVersion("1.0.1.*")]
```
[the remark this answers is not in the document]
Comment (ovelartelius) [reply Addon.Episerver.EnvironmentSynchronizer/Properties/AssemblyInfo.cs:34]: Maybe not. I will lower it to 0.2.1

File: Addon.Episerver.EnvironmentSynchronizer/Addon.Episerver.EnvironmentSynchronizer.csproj [hunk 3]
```diff
@@ -48,6 +48,9 @@
     <Reference Include="EPiServer.Cms.AspNet, Version=11.1.0.0, Culture=neutral, PublicKeyToken=8fe83dea738b45b7, processorArchitecture=MSIL">
       <HintPath>..\packages\EPiServer.CMS.AspNet.11.1.0\lib\net461\EPiServer.Cms.AspNet.dll</HintPath>
     </Reference>
+    <Reference Include="EPiServer.Cms.Shell.UI, Version=11.1.0.0, Culture=neutral, PublicKeyToken=8fe83dea738b45b7, processorArchitecture=MSIL">
```
[the remark this answers is not in the document]
Comment (ovelartelius) [reply Addon.Episerver.EnvironmentSynchronizer/Addon.Episerver.EnvironmentSynchronizer.csproj:51]: I played around with the idea to create one or more supporting admin pages. But dropped that. Maybe later if we have a good reason. Ref removed.

File: Addon.Episerver.EnvironmentSynchronizer/DynamicData/EnvironmentSynchronizationStore.cs [hunk 4]
```diff
@@ -0,0 +1,53 @@
+﻿using System.Linq;
+using EPiServer.Data.Dynamic;
+using EPiServer.Logging;
+using EPiServer.ServiceLocation;
+
+namespace Addon.Episerver.EnvironmentSynchronizer.DynamicData
+{
+	public interface IEnvironmentSynchronizationStore
```
[the remark this answers is not in the document]
Comment (ovelartelius) [reply Addon.Episerver.EnvironmentSynchronizer/DynamicData/EnvironmentSynchronizationStore.cs:8]: All interfaces a in there own file now.

File: Addon.Episever.EnvironmentSynchronizer.Test/app.config [hunk 5]
```diff
@@ -82,6 +82,10 @@
         <assemblyIdentity name="EPiServer.LinkAnalyzer" publicKeyToken="8fe83dea738b45b7" culture="neutral" />
         <bindingRedirect oldVersion="0.0.0.0-11.1.0.0" newVersion="11.1.0.0" />
       </dependentAssembly>
+      <dependentAssembly>
```
[the remark this answers is not in the document]
Comment (ovelartelius) [reply Addon.Episever.EnvironmentSynchronizer.Test/app.config:85]: It is included in the Episerver.CMS.Core nuget package. Do you think we should remove just this reference?

File: Addon.Episerver.EnvironmentSynchronizer/DynamicData/EnvironmentSynchronizationFlag.cs [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+using System.Runtime.Serialization;
+using EPiServer.Data;
+using EPiServer.Data.Dynamic;
+
+namespace Addon.Episerver.EnvironmentSynchronizer.DynamicData
+{
+	[DebuggerDisplay("{Environment} {TimeStamp} {Environment}")]
+	[Serializable]
+	[EPiServerDataStore(AutomaticallyRemapStore = true)]
+	public class EnvironmentSynchronizationFlag : ISerializable, IDynamicData
```
[the remark this answers is not in the document]
Comment (ovelartelius) [reply Addon.Episerver.EnvironmentSynchronizer/DynamicData/EnvironmentSynchronizationFlag.cs:12]: Renamed to *Stamp

**Pull request merged**